ebpf_ffi/ebpf.cc

```diff
@@ -19,57 +19,70 @@ namespace ebpf_ffi {
 // This constant was determined arbitrarily, the number of 0's has incremented
 // when the size was no longer enough for the verifier logs.
 constexpr size_t kLogBuffSize = 100000000;
+// This constnat was determined arbitrarily for the btf logs
+constexpr size_t btfKLogBuffSize = 1024;
 }  // namespace ebpf_ffi
 
-int btf_load(void *btf_buff, size_t btf_size, std::string *error) {
+// btf_buff: Pointer to a buffer where the BTF data is stored
+// btf_size: Size of the BTF data in bytes
+int btf_load(void *btf_buff, size_t btf_size, std::string &error) {
   union bpf_attr btf_attr;
   memset(&btf_attr, 0, sizeof(btf_attr));
   btf_attr.btf = (uint64_t)btf_buff;
   btf_attr.btf_size = btf_size;
 
-  char *btf_log_buf = (char *)malloc(1024);
-  memset(btf_log_buf, 0, 1024);
+  char *btf_log_buf = (char *)malloc(ebpf_ffi::btfKLogBuffSize);
+  memset(btf_log_buf, 0, ebpf_ffi::btfKLogBuffSize);
   btf_attr.btf_log_buf = (uint64_t)btf_log_buf;
-  btf_attr.btf_log_size = 1024;
+  btf_attr.btf_log_size = ebpf_ffi::btfKLogBuffSize;
   btf_attr.btf_log_level = 2;
 
   int btf_fd = syscall(SYS_bpf, BPF_BTF_LOAD, &btf_attr, sizeof(btf_attr));
   if (btf_fd < 0) {
-    *error = strerror(errno);
+    error = strerror(errno);
   }
   return btf_fd;
 }
 
-int load_ebpf_program(uint8_t *prog_buff, int prog_size, uint8_t *btf_buff,
-                      int btf_size, uint8_t *func_buff, int func_size,
-                      std::string *verifier_log, std::string *error) {
+int load_ebpf_program(void *serialized_proto, size_t size,
+                      std::string *verifier_log, std::string &error) {
```

Comment:
you can pass the verifier log as a reference instead of a pointer

Reply:
Done